File: src/cuicui/Meuble.java
package cuicui;

import org.json.*;

public class Meuble {

  private String type_de_meuble;
  private String essence;

  public Meuble(String type_de_meuble,String essence) {
    this.type_de_meuble = type_de_meuble;
    this.essence = essence;

  }

  @Override
  public String toString() {
    return "Meuble : " + this.type_de_meuble + ", " + this.essence;
  }

  public JSONObject toJSON() {

    var json = new JSONObject();

    json.put("essence", this.essence);
    json.put("type_de_meuble", this.type_de_meuble);

    return json;
  }

  public static Meuble fromJSON(String json) {
    JSONObject jo = new JSONObject(json);

    String type_m = jo.getString("type_de_meuble");

    String ess = jo.getString("essence");

    return new Meuble(type_m, ess);
  }

  public String getTypeMeuble() { return this.type_de_meuble; }
  public String getEssence() { return this.essence; }

}

File: PHP/includes/Meuble.php
<?php
class Meuble implements JSONSerializable{
    private string $type_de_meuble;
    private string $essence;

    public function getTypeDeMeuble():string{
        return $this->type_de_meuble;
    }

    public function getEssence():string{
        return $this->essence;
    }
    public function __construct(string $type_de_meuble,string $essence){
        $this->type_de_meuble=$type_de_meuble;
        $this->essence=$essence;
    }
    public function __toString():string{
        return $this->type_de_meuble." ".$this->essence;

    }
    public function jsonSerialize():array{
        return["type_de_meuble"=>$this->getTypeDeMeuble(),
        "essence"=>$this->getEssence()];
    }
    public static function fromJson(string $json):Meuble{
        $array=json_decode($json,true);
        return new Meuble($array["type_de_meuble"],$array["essence"]);
    }
}
